File: app/src/main/java/com/example/realm_template/Migration.java
package com.example.realm_template;

import io.realm.DynamicRealm;
import io.realm.Realm;
import io.realm.RealmMigration;

public class Migration implements RealmMigration {
    // increment this if schema changed
    public static final long SCHEMA_VERSION = 0;

    @Override
    public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {

        //if (oldVersion == 0) {
        //    // migrate to 1
        //
        //    // write migration code here
        //
        //    oldVersion++;
        //}

        if (oldVersion != SCHEMA_VERSION) {
            throw new RuntimeException("unexpected scheme version. expected: " + SCHEMA_VERSION + ", actual: " + oldVersion);
        }
    }
}

File: kotlin_app/src/main/kotlin/com/example/realm_kotlin_template/Migration.kt
package com.example.realm_kotlin_template

import io.realm.DynamicRealm
import io.realm.Realm
import io.realm.RealmMigration

class Migration : RealmMigration {

    override fun migrate(realm: DynamicRealm?, oldVersion: Long, newVersion: Long) {
        //if (oldVersion == 0) {
        //    // migrate to 1
        //
        //    // write migration code here
        //
        //    oldVersion++;
        //}

        if (oldVersion != SCHEMA_VERSION) {
            throw RuntimeException("unexpected scheme version. expected: $SCHEMA_VERSION, actual: $oldVersion")
        }
    }

    companion object {
        // increment this if schema changed
        val SCHEMA_VERSION: Long = 0
    }
}
